### PR TITLE
16720 - integrate NR search

### PR DESCRIPTION
**Pull request by Jxio**
*Issue #:*
https://github.com/bcgov/entity/issues/16720

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ BC Registries authentication and authorization interface.
 
 ## Prerequisites
 - Node.js (> v13.6.x)
```
Comment (seeker25) [inline auth-web/README.md:12]: Node 16.15.x

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue [hunk 2]
```diff
@@ -144,124 +144,126 @@
...
-            class="new-actions mx-auto"
-          >
-            <!--  tech debt ticket to improve this piece of code. https://github.com/bcgov/entity/issues/17132 -->
-            <span class="open-action">
-              <v-tooltip
-                top
-                content-class="top-tooltip"
-                :disabled="disableTooltip(item)"
+        <!-- Status -->
+        <template v-slot:item-slot-Status="{ item }">
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:148]: Remove the slot above, we dont need it, it's duplicated

File: auth-web/src/components/auth/manage-business/BusinessLookup.vue [hunk 3]
```diff
@@ -96,61 +131,82 @@ enum States {
...
-      // enable or disable items according to whether they have already been added
-      for (const result of this.searchResults) {
-        result.disabled = await this.isAffiliated(result.identifier)
+// @Component({})
+export default defineComponent({
+  name: 'BusinessLookup',
+  props: {
+    lookupType: { type: String, default: LookupType.BUSINESS }
+  },
+  setup (props, { emit }) {
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/BusinessLookup.vue:140]: I think usually we use `emits: ['']`

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2464-3hojj7lz.web.app

File: auth-web/src/components/auth/manage-business/BusinessLookup.vue (hunk 3 above)
Comment (Jxio) [reply auth-web/src/components/auth/manage-business/BusinessLookup.vue:140]: I checked other components, they used { emit } as well, type error if change to emits: ['']

Comment (seeker25) [reply auth-web/src/components/auth/manage-business/BusinessLookup.vue:140]: You need to use both, otherwise you get a warning when you emit

Comment (seeker25) [reply auth-web/src/components/auth/manage-business/BusinessLookup.vue:140]: image

Comment (seeker25) [reply auth-web/src/components/auth/manage-business/BusinessLookup.vue:140]: https://vuejs.org/guide/components/events.html#declaring-emitted-events

Comment (seeker25) [reply auth-web/src/components/auth/manage-business/BusinessLookup.vue:140]: image

File: auth-web/src/services/name-request-lookup.services.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  static async search (query: string): Promise<NameRequestLookupResultIF[]> {
+    let url = this.namexApiUrl + 'requests/search'
+    url += `?query=${encodeURIComponent(query)}`
+    url += '&start=0&rows=20'
+    const token = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken)
+
+    try {
+      const response = await axios.get(url, {
+        headers: {
+          'Authorization': `Bearer ${token}`,
```
Comment (seeker25) [inline auth-web/src/services/name-request-lookup.services.ts:31]: don't the built in interceptors already handle this?

File: auth-web/src/services/name-request-lookup.services.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          'Content-Type': 'application/json'
+        }
+      })
+      console.log(response)
```
Comment (seeker25) [inline auth-web/src/services/name-request-lookup.services.ts:35]: console.log

File: auth-web/src/services/name-request-lookup.services.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+   */
+  static async search (query: string): Promise<NameRequestLookupResultIF[]> {
+    let url = this.namexApiUrl + 'requests/search'
+    url += `?query=${encodeURIComponent(query)}`
+    url += '&start=0&rows=20'
+    const token = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken)
+
+    try {
+      const response = await axios.get(url, {
+        headers: {
```
[the remark this answers is not in the document]
Comment (seeker25) [reply auth-web/src/services/name-request-lookup.services.ts:30]: already have interceptors do this?

File: auth-web/src/components/auth/manage-business/BusinessLookup.vue [hunk 7]
```diff
@@ -96,61 +131,82 @@ enum States {
...
+      // safety check
+      if (searchField.value && searchField.value?.length < 3) {
+        state.value = States.TYPING
+      } else if (searchField.value && searchField.value?.length > 2) {
+        state.value = States.SEARCHING
+        const searchStatus = null // search all (ACTIVE + HISTORICAL)
+
+        // Use appropriate service based on lookupType
+        if (props.lookupType === LookupType.NR) {
+          searchResults.value = await NameRequestLookupServices.search(searchField.value).catch(() => [])
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/BusinessLookup.vue:166]: Catch with no handling?

File: auth-web/src/components/auth/manage-business/BusinessLookup.vue [hunk 8]
```diff
@@ -96,61 +131,82 @@ enum States {
...
+        } else {
+          searchResults.value = await BusinessLookupServices.search(searchField.value, searchStatus).catch(() => [])
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/BusinessLookup.vue:168]: Catch with no handling?

File: auth-web/src/components/auth/manage-business/BusinessLookup.vue [hunk 9]
```diff
@@ -96,61 +131,82 @@ enum States {
...
-        result.disabled = await this.isAffiliated(result.identifier)
+// @Component({})
+export default defineComponent({
+  name: 'BusinessLookup',
+  props: {
+    lookupType: { type: String, default: LookupType.BUSINESS }
+  },
+  setup (props, { emit }) {
+    // local variables
+    const state = ref(States.INITIAL)
```
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/BusinessLookup.vue:142]: Probably could use `reactive`

File: auth-web/src/components/auth/manage-business/BusinessLookup.vue (hunk 3 above)
Comment (Jxio) [reply auth-web/src/components/auth/manage-business/BusinessLookup.vue:141]: updated

File: auth-web/src/services/name-request-lookup.services.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Jxio) [reply auth-web/src/services/name-request-lookup.services.ts]: yes, I just removed headers

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2464-3hojj7lz.web.app

**Review by seeker25 — APPROVED** [review]

**Pull request merged**